### PR TITLE
Add vapid-web-push extension

### PR DESCRIPTION
**Pull request by iksnerd**
## Checklist

- [ ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Why do you have email in preferences for both `Generate VAPID Keys` and for the extension itself?

- The command Send Puh Notification should use `useForm` from `@raycast/api` instead, preferences is not the place 🙂

- Can you record a screencast of the flow and how the action works?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

File: extensions/vapid-web-push/src/setup.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  email: string;
+  publicKey: string;
+  privateKey: string;
+  endpoint: string;
+  p256dh: string;
+  auth: string;
+  title: string;
+  body: string;
+}
+
```
Comment (pernielsentikaer) [inline extensions/vapid-web-push/src/setup.tsx:13]:
```suggestion

```
Already a part of raycast-env.d.ts

**Comment by iksnerd** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * Why do you have email in preferences for both `Generate VAPID Keys` and for the extension itself?
> * The command Send Puh Notification should use `useForm` from `@raycast/api` instead, preferences is not the place 🙂
> * Can you record a screencast of the flow and how the action works?
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I will check those and get back to you. Thanks for the feedback

File: extensions/vapid-web-push/src/setup.tsx (hunk 1 above)
Comment (iksnerd) [reply extensions/vapid-web-push/src/setup.tsx]: I think I made the usage more obvious. Let me know if I have to fix something else.

**Comment by pernielsentikaer** [comment]
Replied on slack 🙂

**Comment by pernielsentikaer** [comment]
Can we handle error gracefully? It seems to crash with wrong credentials provided

vapid-web-push 2024-10-09 at 12 24 48

Can you record a screencast of the flows?

**Pull request closed without merging**